editor.planx.uk/src/@planx/components/shared/DataFieldAutocomplete.tsx

```diff
@@ -0,0 +1,99 @@
+import {
+  AutocompleteChangeReason,
+  AutocompleteProps,
+  createFilterOptions
+} from "@mui/material/Autocomplete";
+import ListItem from "@mui/material/ListItem";
+import isNull from "lodash/isNull";
+import React, { useMemo } from "react";
+import AutocompleteInput from "ui/shared/AutocompleteInput";
+import InputRow from "ui/shared/InputRow";
+
+interface Props {
+  schema?: string[];
+  value?: string;
+  onChange: (value: string | null) => void;
+  required?: boolean;
+}
+
+const renderOptions: AutocompleteProps<
+  string,
+  false,
+  false,
+  true,
+  "div"
+>["renderOption"] = (props, option) => {
+  return (
+    <ListItem key={option} sx={{ fontFamily: (theme) => theme.typography.data.fontFamily }} {...props}>
+      {option}
+    </ListItem>
+  );
+};
+
+const filter = createFilterOptions<string>();
+
+export const DataFieldAutocomplete: React.FC<Props> = (props) => {
+  const { value: initialValue, schema: options = [] } = props;
+
+  const value: string | undefined = useMemo(
+    () => options?.find((option) => option === initialValue),
+    [initialValue, options],
+  );
+
+  const handleChange = (
+    _event: React.SyntheticEvent,
+    value: string | null,
+    _reason: AutocompleteChangeReason,
+  ) => {
+    if (typeof value === "string") {
+      // Adding a new option
+      if (value.startsWith('Add "')) {
+        props.onChange(value.split(('"'))[1]);
+      } else {
+        // Selecting an option
+        props.onChange(value);
+      }
+    } else if (isNull(value)) {
+      // Clearing an option
+      props.onChange(value);
+    }
+  };
+
+  return (
+    <InputRow>
+      <AutocompleteInput
+        id="data-field-autocomplete"
+        key="data-field-autocomplete"
+        placeholder="Data field"
+        required={Boolean(props.required)}
```

Comment:
`required` will simply trigger browser default error handling which is quite consistent with current editor inputs ! Gov UK-looking error wrappers for autocomplete are larger future thing to revisit (eg would we simply wrap in our existing wrapper or account for new `error`-related props here?)
![Screenshot from 2024-12-18 10-13-06](https://github.com/user-attachments/assets/d2d427a4-87c9-41cc-b490-4442c7530f30)